### PR TITLE
Remove mlflow.keras raise deprecation warning to avoid keras==2.4.3 tensorflow==2.5.1 issue

### PR DESCRIPTION
**Pull request by eddiestudies**
## What changes are proposed in this pull request?

Adding a protection against attribute errors that arise when tensorflow.keras and keras are both installed.
https://stackoverflow.com/questions/61137954/attributeerror-module-tensorflow-python-keras-utils-generic-utils-has-no-attr

keras==2.4.3
tensorflow==2.5.1

## How is this patch tested?

The above error is now silent in the original code path. The keras_module is set to tensorflow.keras and successfully logs the model

## Release Notes

Patch bug due to incompatible keras and tensorflow.keras versions. 

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by sidhartha-roy — APPROVED** [review]
looks good

**Comment by eedeleon** [comment]
@trangevi  I ran into an issue with keras import failing for the provided versions

../../../.local/lib/python3.8/site-packages/mlflow/keras.py:404: in log_model
    Model.log(
../../../.local/lib/python3.8/site-packages/mlflow/models/model.py:187: in log
    flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
../../../.local/lib/python3.8/site-packages/mlflow/keras.py:193: in save_model
    _raise_deprecation_warning()
../../../.local/lib/python3.8/site-packages/mlflow/keras.py:68: in _raise_deprecation_warning
    import keras
../../../.local/lib/python3.8/site-packages/keras/__init__.py:20: in <module>
    from . import initializers
../../../.local/lib/python3.8/site-packages/keras/initializers/__init__.py:124: in <module>
    populate_deserializable_objects()
    
    AttributeError: module 'keras.utils.generic_utils' has no attribute 'populate_dict_with_module_objects'

../../../.local/lib/python3.8/site-packages/keras/initializers/__init__.py:82: AttributeError

File: mlflow/keras.py [hunk 1]
```diff
@@ -61,9 +61,11 @@
 _PIP_ENV_SUBPATH = "requirements.txt"
 
 
-def _raise_deprecation_warning():
+def _raise_deprecation_warning(keras_module=None):
```
Comment (harupy) [inline mlflow/keras.py:64]: @eedeleon We're planning to remove this function and drop support for keras < 2.3.0 and tensorflow < 2.0.

Comment (eedeleon) [reply mlflow/keras.py:64]: Makes sense. However, currently I get a failure for 

keras==2.4.3
tensorflow==2.5.1

Would the correct PR be to remove it altogether?

Comment (harupy) [reply mlflow/keras.py:64]: @eedeleon Thanks for the reply. Got it. If we `mlflow.keras._raise_deprecation_warning` that contains `import keras`, does that solve the issue?

File: mlflow/R/mlflow/DESCRIPTION [hunk 2]
```diff
@@ -61,7 +61,7 @@ Suggests:
     testthat (>= 2.0.0),
     xgboost
 Encoding: UTF-8
-RoxygenNote: 7.1.1
+RoxygenNote: 7.1.2
```
Comment (harupy) [inline mlflow/R/mlflow/DESCRIPTION:64]: Can we revert this change?

Comment (eedeleon) [reply mlflow/R/mlflow/DESCRIPTION]: I am not sure why it was added to my branch but I reverted it

File: mlflow/keras.py (hunk 1 above)
Comment (eedeleon) [reply mlflow/keras.py:64]: I think you mean remove. If so, yes

Comment (harupy) [reply mlflow/keras.py:64]: Sorry, "remove" was missing.

Comment (eedeleon) [reply mlflow/keras.py:64]: Would you like me to make a different PR removing the function and all calls to it? Or is someone else doing so.

Comment (harupy) [reply mlflow/keras.py:64]: That'd be great!

**Review by harupy — APPROVED** [review]
LGTM!
